pkg/apis/externaldns/validation/validation.go

```diff
@@ -42,5 +42,21 @@ func ValidateConfig(cfg *externaldns.Config) error {
 			return errors.New("no Azure config file specified")
 		}
 	}
+
+	validLogLevels := map[string]bool{
+		"debug": true,
+		"info":  true,
+		"warn":  true,
+		"error": true,
+		"fatal": true,
+	}
+	if _, ok := validLogLevels[cfg.LogLevel]; !ok {
+		return fmt.Errorf("unsupported log level: %s", cfg.LogLevel)
+	}
+
+	if cfg.Debug && cfg.LogLevel != "debug" {
+		return fmt.Errorf("conflicting log levels, --debug and --log-level=%s", cfg.LogLevel)
+	}
```

Comment:
I would be fine with breaking backwards compatibility in order to remove this check. What do you think @ideahitme?

Reply:
agree, please drop `--debug` flag

Reply:
Done.